opm/simulators/flow/python/PyBaseSimulator.hpp

```diff
@@ -0,0 +1,115 @@
+/*
+  Copyright 2020 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef OPM_PY_BASE_SIMULATOR_HEADER_INCLUDED
+#define OPM_PY_BASE_SIMULATOR_HEADER_INCLUDED
+
+#include <opm/models/utils/parametersystem.hpp>
+#include <opm/models/utils/propertysystem.hh>
+
+#include <opm/simulators/flow/FlowMain.hpp>
+#include <opm/simulators/flow/python/PyMain.hpp>
+#include <opm/simulators/flow/python/PyFluidState.hpp>
+#include <opm/simulators/flow/python/PyMaterialState.hpp>
+#include <opm/simulators/flow/python/Pybind11Exporter.hpp>
+
+#include <opm/input/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.hpp>
+#include <opm/input/eclipse/Schedule/Schedule.hpp>
+#include <opm/input/eclipse/Deck/Deck.hpp>
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Opm::Pybind {
+
+template<class TypeTag>
+class PyBaseSimulator
+{
+private:
+    using Simulator = Opm::GetPropType<TypeTag, Opm::Properties::Simulator>;
+
+public:
+    PyBaseSimulator(const std::string& deckFilename,
+        const std::vector<std::string>& args);
+    PyBaseSimulator(
```

Comment:
clang-format has barfed all over this code I see. I'll not comment on the others..